### ISSUE TITLE
Admin Section is not visible in backend on production mode

### ISSUE THREAD

**Issue by cdiacon** [magento/magento2#14104]
I cannot access the section if the section id or group contains "dev" as an id and the production mode is enabled as a result the admin visibility will change the section to hidden.

### Preconditions

1. magento with the branch 2.2.4-preview installed
2. php 7.0
3. production mode

### Steps to reproduce

1. create a section in adminhtml/system.xml in a custom module with the section id that contains "dev" 
2. enable the production mode
3. go to stores->configuration 
4. the section is not visible 

### Expected result
1. have a section/group with the id that conains "dev" as a section id example
```
<section id="anything_that_contains_dev_as_a_name_will_be_hidden" translate="label" sortOrder="130" showInDefault="1" showInWebsite="1" showInStore="0">
```
2. enable the production mode
3. see the section visible on stores->configuration

### Actual result

1. section is not visible 

![screen shot 2018-03-14 at 16 18 26](https://user-images.githubusercontent.com/728734/37415880-5f60ba48-27a4-11e8-9547-e29a756696cb.png)
![screen shot 2018-03-14 at 16 30 51](https://user-images.githubusercontent.com/728734/37416255-333c00ca-27a5-11e8-9f76-7ca63fb072cd.png)

**Comment by cdiacon** [comment]
found the issue that may be related to this change here

https://github.com/magento/magento2/pull/12207

So this Class https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Config/Model/Config/Structure/ConcealInProductionConfigList.php#L81

the ishidden method was changed from 

```
 public function isHidden($path)
    {
        $path = $this->normalizePath($path);
        return $this->state->getMode() === State::MODE_PRODUCTION
            && !empty($this->configs[$path])
            && $this->configs[$path] === static::HIDDEN;
    }
```
to 

```
 public function isHidden($path)
    {
        $result = false;
        $path = $this->normalizePath($path);
        if ($this->state->getMode() === State::MODE_PRODUCTION
            && preg_match('/(?<group>(?<section>.*?)\/.*?)\/.*?/', $path, $match)) {


            $group = $match['group'];
            $section = $match['section'];
            $exemptions = array_keys($this->exemptions);
            foreach ($this->configs as $configPath => $value) {
                if ($value === static::HIDDEN && strpos($path, $configPath) !==false) {
                    $result = empty(array_intersect([$section, $group, $path], $exemptions));
                }
            }
        }

        return $result;
    }
```

this line here

if ($value === static::HIDDEN && strpos($path, $configPath) !==false) {


using the strpos to find anything that in this case $configPath = "dev" will be hidden

like
"anythingdevname"
"devanyname"
"anynamedev"
etc...

**Comment by fooman** [comment]
@cdiacon agreed that this is indeed somewhat unexpected behaviour - however looking at https://github.com/magento/magento2/pull/12207/files#diff-94a0fc0a8eaeff073c1bf44718066492R148 it should be possible for you to add an exemption.

**Comment by magento-engcom-team** [comment]
@cdiacon, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @cdiacon. Thank you for your report. 
The issue has been fixed in magento/magento2#15174 by @hostep in 2.2-develop branch
Related commit(s):
 - [20def1ee0124d163f57912006b86094fb5835c8d](https://github.com/magento/magento2/commit/20def1ee0124d163f57912006b86094fb5835c8d)

The fix will be available with the upcoming 2.2.5 release.

**Comment by hostep** [comment]
@cdiacon: please ignore the comment above, your issue was already fixed in 2.2.4 :)
